internal/policy/container/trademark_validator.go

```diff
@@ -0,0 +1,20 @@
+package container
+
+import (
+	"regexp"
+	"strings"
+)
+
+// ViolatesRedHatTrademark validates if a string meets specific "Red Hat" naming criteria
+func ViolatesRedHatTrademark(nameLabel string) bool {
+	// string starts with Red Hat variant
+	startingWithRedHat := regexp.MustCompile("^[^a-z0-9]*red[^a-z0-9]*hat").MatchString(strings.ToLower(nameLabel))
+
+	// string contain Red Hat variant (not starting with)
+	containsRedHat := regexp.MustCompile("red[^a-z0-9]*hat").MatchString(strings.ToLower(nameLabel))
```

Comment:
Do we care about `redhat` (all one word)? I realize it's probably not a specific trademark violation, but can't hurt to ask.

Reply:
I think we do, and a test would fail with this string.

Reply:
I added a test for `redhat`, let me know what you think.